src/main/java/iudx/resource/server/database/elastic/ElasticClient.java

```diff
@@ -154,6 +156,27 @@ public Future<JsonObject> asyncScroll(
     return promise.future();
   }
 
+  private void convertToGivenType(String format, List<Hit<ObjectNode>> searchHits) {
+    if(format.equals(HEADER_CSV))
+    {
+      convertToCSV(searchHits);
+    }
+    else if(format.equals(HEADER_PARQUET))
+    {
+      LOGGER.debug("Convert from Elastic response to parquet format");
+    }
+    else
+    {
+      LOGGER.debug("Get the elastic response in JSON format");
+    }

```

Comment:
Instead of using If-else for every write request, use a Strategy pattern to decide which file writing strategy to use and then call `append()` or `write()` directly on that strategy object.

Reply:
Updated it here : [Update files](https://github.com/datakaveri/iudx-resource-server/pull/425/commits/406ad40d4131080152390c3e65eba43b2d055360)